libraries/ZigBeeNet/Util/ByteHelper.cs

```diff
@@ -37,5 +39,33 @@ public static long LongFromBytes(byte[] values, int msb, int lsb)
 
             return result;
         }
+
+        public static byte[] Slice(this byte[] data, int offset, int len)
+        {
+            if (offset < 0)
+                throw new ArgumentOutOfRangeException(nameof(offset));
+            if (len < 0)
+                throw new ArgumentOutOfRangeException(nameof(len));
+            if (data.Length < offset + len)
+                throw new ArgumentException("Not enough data for the slice");
+            byte[] range = new byte[len];
+            Buffer.BlockCopy(data, offset, range, 0, len);
+            offset += len;
```

Comment:
Hi,

Whats this line for?

regards

Reply:
It's an oversight, it should be removed. In the original code I'm using in another project the function was passing `ref int offset` with the goal to update the original offset with the len.

Reply:
You should be able to commit on your develop branch, so that it will be included in this PR, too. Threre should also be a hint in this PR below.
I would also reccommend you to create a feature branch in the feature, so PRs won't contain other changes

Reply:
We could also consider using the Ranges introduced with C# 8. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/ranges. But may be for the next step :)

Reply:
`Span<byte>` and `ReadOnlySpan<byte>` are also very useful in these scenarios. But for Range you need either target NetStandard2.1 or used an unsupported hack (https://www.meziantou.net/how-to-use-csharp-8-indices-and-ranges-in-dotnet-standard-2-0-and-dotn.htm).